robot_controll/main.cpp

```diff
@@ -206,39 +207,42 @@ CvScalar status_to_color(string status) {
 
     return it->second;
 }
-
+ 
 /*
  *
  */
 
 int main(int argc, char** argv) {
-    enum image_source_type {camera, jpg, avi};
+    
+    enum image_source_type {
+        camera, jpg, avi
+    };
     image_source_type image_source = camera;
     string image_file = "file.jpg";
-    enum run_mode_type {online, offline};
+
+    enum run_mode_type {
+        online, offline
+    };
     run_mode_type run_mode = online;
-    
+
     IplImage* localizationFrame;
-    YAML::Node config = YAML::LoadFile("../config.yaml");
-    
-    if (config["image_source"].as<string>().compare("camera") == 0)
+
+    if (Config::getString("image_source").compare("camera") == 0)
```

Comment:
@D0d0 if you feel like it, I believe we could simplify this way way more.

For instance, we could just look at the extension (last four characters) of `image_source`. If it was `.avi` it would be `avi` and `image_file` would become `image_source`. The same would happen in the case of `.jpg`. Otherwise, `image_source` would be the camera.

I strongly doubt we'd need this much spaghetti then.